Sources/TensorFlow/Epochs/Sampling.swift

```diff
@@ -46,18 +46,33 @@ extension LazilySelected: Collection {
   public func index(after i: Index) -> Index { selection.index(after: i) }
 }
 
-extension LazilySelected: BidirectionalCollection
+extension Sampling: BidirectionalCollection
 where Selection: BidirectionalCollection {
   /// Returns the position after `i`.
   public func index(before i: Index) -> Index { selection.index(before: i) }
 }
 
-extension LazilySelected: RandomAccessCollection
+extension Sampling: RandomAccessCollection
 where Selection: BidirectionalCollection {
   /// Returns the position `n` places from `i`.
   public func index(_ i: Index, offsetBy n: Int) -> Index {
     selection.index(before: i)
   }
+  
+  // Needed because of https://bugs.swift.org/browse/SR-12692
+  /// Returns `i` offset by `distance` unless that requires passing `limit`, in 
+  /// which case `nil` is returned.
```

Comment:
I'm really not sure what to do about SR-12692, since as you can see by the latest comments, the problem is unfixable.  We need a project policy.

Reply:
I'm not sure how to take action on this, do you want to remove the mention to the bug?